src/test/compile-fail/E0204.rs

```diff
@@ -9,19 +9,35 @@
 // except according to those terms.
 
 struct Foo {
-    foo: Vec<u32>,
+    foo: Vec<u32>, //~ this field does not implement `Copy`
 }
 
 impl Copy for Foo { }
 //~^ ERROR E0204
-//~| NOTE field `foo` does not implement `Copy`
 
 #[derive(Copy)]
 //~^ ERROR E0204
-//~| NOTE field `ty` does not implement `Copy`
-//~| NOTE in this expansion of #[derive(Copy)]
 struct Foo2<'a> {
-    ty: &'a mut bool,
+    ty: &'a mut bool, //~ this field does not implement `Copy`
+}
+
+enum EFoo {
+    Bar {
+        x: Vec<u32> //~ this field does not implement `Copy`
+    },
+    Baz,
+}
+
+impl Copy for EFoo { }
+//~^ ERROR the trait `Copy` may not be implemented for this type
+
+#[derive(Copy)]
+//~^ ERROR the trait `Copy` may not be implemented for this type
+enum EFoo2<'a> {
+    Bar(
+        &'a mut bool //~ this field does not implement `Copy`
+    ),
+    Baz,
```

Comment:
can we make this a `ui` test? I'd like to see precisely where the span is, since it's critical to the message